### PR TITLE
GPU Kernel Conditional Statement Bug Fix

### PR DESCRIPTION
**Pull request by ShreyasKhandekar**
Fixing a bug in the compiler where if a GPUizable loop has a conditional statement whose expression comes from outside the loop, we do not add it as a argument to the code generated for the kernel.

Added conditions in `gpuTransforms.cpp` so that it adds that expression as a kernel argument.

Also added a catch-all `else` block that causes an error if there happen to be any other such cases where we forget to add a compiler argument.

Added a test which fails on main but passes with this fix.

- [X] Local test
- [ ] Paratests

### CODE REVIEW

**Review by stonea — COMMENTED** [review]
Can you check that the `INT_FATAL("Unexpected symbol expression");` doesn't get triggered by any of our existing tests? Just run `start_tests` under `test/gpu/native` on Horizon or Osprey and make sure nothing fails.

If that passes I'm good with merging this.


Note there's a couple of test failures we have from last night so these two are expected: https://chapel.discourse.group/t/cron-cray-cs-gpu-native/14597

File: test/gpu/native/innerBlock.chpl [hunk 1]
```diff
@@ -60,6 +60,22 @@ on here.getChild(1) {
   writeArr(A);
 
 }
+
+// Create test for arg bug
```
Comment (e-kayrakli) [inline test/gpu/native/innerBlock.chpl:64]: drop comment

File: test/gpu/native/innerBlock.chpl [hunk 2]
```diff
@@ -60,6 +60,22 @@ on here.getChild(1) {
...
+on here.gpus[0] {
+  var A = [1,2,3,4,5];
+  var outerVar = true;
+
+    forall a in A {
```
Comment (e-kayrakli) [inline test/gpu/native/innerBlock.chpl:69]: dedent this loop

**Review by e-kayrakli — APPROVED** [review]
Thanks Shreyas!

I hope the tests will pass. If it doesn't we probably revert that INT_FATAL into just a comment for now saying that we may need to special case some things down the road there, or we may need to generalize the implementation a little bit better. But we don't know what those special cases are yet and it is a bit too early to think about refactoring here. For now we are just directly copying nodes which contains symbols for which we don't have a special action.

**Pull request merged**